test/Microsoft.Azure.Devices.Edge.Test/Module.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Test.Common;
+    using Microsoft.Azure.Devices.Edge.Test.Common.Config;
+    using Microsoft.Azure.Devices.Edge.Test.Helpers;
+    using NUnit.Framework;
+
+    public class Module : ModuleBase
+    {
+        [Test]
+        public async Task TempSensor()
+        {
+            string sensorImage = Context.Current.TempSensorImage.Expect(() => new ArgumentException());
+
+            CancellationToken token = this.cts.Token;
+
+            EdgeDeployment deployment = await this.runtime.DeployConfigurationAsync(
```

Comment:
Where does edge daemon get deployed?

Reply:
Are these tests run after edge daemon get deployed?

Reply:
In the setup fixture (see `SetupFixture.cs`). Because of the `[OneTimeSetUp]` attribute, NUnit runs this code once at the start of the test run.

Reply:
> Are these tests run after edge daemon get deployed?

Yes.

Reply:
Do you mean SetupFixure will be run once for each tests in Module class? Will SetupFixure be run for each test in Device class? if not, how this be configured?

Reply:
No. For a full test run, here's what gets called:

```
Edge.Test.SetupFixture.BeforeAllAsync()

  Edge.Test.Helpers.DeviceBase.BeforeAllAsync()

    Edge.Test.Helpers.TestBase.BeforeEachTest()
      Edge.Test.Device.TransparentGateway(SasInScope, Mqtt)
    Edge.Test.Helpers.TestBase.AfterEachTestAsync()
    
    <9 remaining permutations of TransparentGateway test...>

  Edge.Test.Helpers.DeviceBase.AfterAllAsync()

  Edge.Test.Helpers.TestBase.BeforeEachTest()
    Edge.Test.Helpers.ModuleBase.BeforeEachModuleTest()
      Edge.Test.Module.ModuleToModuleDirectMethod(Amqp)
  Edge.Test.Helpers.TestBase.AfterEachTestAsync()

  <3 remaining permutations of ModuleToModuleDirectMethod test...>

  Edge.Test.Helpers.TestBase.BeforeEachTest()
    Edge.Test.Helpers.ModuleBase.BeforeEachModuleTest()
      Edge.Test.Module.TempSensor()
  Edge.Test.Helpers.TestBase.AfterEachTestAsync()

Edge.Test.SetupFixture.AfterAllAsync()
```

You can see NUnit docs on `OneTimeSetUp`, `OneTimeTearDown`, `SetUp`, and `TearDown` attributes for more info. https://github.com/nunit/docs/wiki/Attributes

Reply:
Both SetupFixture and DeviceBase assigns this.daemon; why?